### PR TITLE
build: update client-go and other kube dependencies to 1.20.6

### PR DESCRIPTION
**Pull request by humblec**
build: update client-go and other kube dependencies to 1.20.6
    
    client-go 1.20.6 has a fix for below CVE: This patch address this
    via updating client-go and other dependencies.
    
    CVE-2019-11250 : The MITRE CVE dictionary describes this issue as:
    
    The Kubernetes client-go library logs request headers at verbosity
    levels of 7 or higher. This can disclose credentials to unauthorized
    users via logs or command output. Kubernetes components (such as
    kube-apiserver) prior to v1.16.0, which make use of basic or bearer
    token authentication, and run at high verbosity levels, are affected.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -26,7 +26,7 @@ require (
 	google.golang.org/grpc v1.36.1
 	k8s.io/api v0.20.0
 	k8s.io/apimachinery v0.20.0
-	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
+	k8s.io/client-go v0.20.6-rc.0
```
Comment (Madhu-1) [inline go.mod]: why use RC instead of the release version?

Comment (humblec) [reply go.mod]: Yeah, better to use release version. updated.. Thanks @Madhu-1  ptal

Comment (humblec) [reply go.mod]: @Madhu-1 rather I think its good to bring the other kube dependency to latest with client-go to be in parity . wdyt?

**Review by nixpanic — DISMISSED** [review]
add a link to the CVE in the commit message. MITRE is an organization/database that manages CVEs, the actual CVE that is addressed is important to know when stating a fix for one

**Comment by humblec** [comment]
> add a link to the CVE in the commit message. MITRE is an organization/database that manages CVEs, the actual CVE that is addressed is important to know when stating a fix for one

Missed to add CVE number, I have udpated the PR with the same. PTAL. Thanks @nixpanic

**Comment by nixpanic** [comment]
> Missed to add CVE number, I have udpated the PR with the same. PTAL. Thanks @nixpanic

Please add it to the commit message as well, not only the PR description. Ideally include a link to the CVE too.

**Comment by nixpanic** [comment]
Is there a reason to move to Kubernetes 1.20, and not the current 1.21?

**Comment by Rakshith-R** [comment]
> Is there a reason to move to Kubernetes 1.20, and not the current 1.21?

It cannot be updated to 1.21 yet. I kinda already did it at https://github.com/ceph/ceph-csi/pull/1974 but we need to wait for controller-runtime v0.9.0 release, see https://github.com/kubernetes-sigs/controller-runtime/issues/1473.

**Comment by humblec** [comment]
@nixpanic @Madhu-1 ptal. Thanks.

**Comment by humblec** [comment]
@Madhu-1 @nixpanic ptal.

**Review by Madhu-1 — DISMISSED** [review]
This is not only updating the client-go it is updating most of the kubernetes dependency from 0.20.0 to 0.20.6 can you please update the commit message and PR title?

**Comment by Madhu-1** [comment]
@humblec please add a link to the client.go release in which this is fixed

**Comment by humblec** [comment]
> This is not only updating the client-go it is updating most of the kubernetes dependency from 0.20.0 to 0.20.6 can you please update the commit message and PR title?

Updated commit and PR description.

**Review by Madhu-1 — DISMISSED** [review]

**Review by nixpanic — DISMISSED** [review]

**Comment by humblec** [comment]
@Madhu-1 @nixpanic can you please bring back the approval here? Thanks

File: go.mod [hunk 2]
```diff
@@ -24,13 +24,13 @@ require (
...
-	k8s.io/api v0.20.0
-	k8s.io/apimachinery v0.20.0
-	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
-	k8s.io/cloud-provider v0.20.0
+	k8s.io/api v0.20.6
+	k8s.io/apimachinery v0.20.6
+	k8s.io/client-go v0.20.6
+	k8s.io/cloud-provider v0.20.6
 	k8s.io/klog/v2 v2.4.0
 	k8s.io/kubernetes v1.20.0
```
Comment (Madhu-1) [inline go.mod]: can you also update `k8s.io/kubernetes` to v1.20.6?

Comment (humblec) [reply go.mod]: Done. thanks .

**Comment by humblec** [comment]
@Madhu-1 @nixpanic resolved conflict and rebased to devel. PTAL

**Review by Madhu-1 — APPROVED** [review]

**Review by nixpanic — APPROVED** [review]

**Comment by humblec** [comment]
@mergifyio rebase

**Comment by Rakshith-R** [comment]
> /retest ci/centos/mini-e2e/k8s-1.19

```
! StartHost failed, but will try again: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute

* Creating kvm2 VM (CPUs=8, Memory=14336MB, Disk=20000MB) ...

* Failed to start kvm2 VM. Running "minikube delete" may fix it: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute



X Exiting due to GUEST_PROVISION: Failed to start host: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute
```

**Comment by humblec** [comment]
> > /retest ci/centos/mini-e2e/k8s-1.19
> 
> ```
> ! StartHost failed, but will try again: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute
> 
> * Creating kvm2 VM (CPUs=8, Memory=14336MB, Disk=20000MB) ...
> 
> * Failed to start kvm2 VM. Running "minikube delete" may fix it: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute
> 
> 
> 
> X Exiting due to GUEST_PROVISION: Failed to start host: creating host: create: Error creating machine: Error in driver during machine creation: IP not available after waiting: machine minikube didn't return IP after 1 minute
> ```

@Rakshith-R  Yeah!!  I have been waiting for getting this merged since yesterday.. :(

**Pull request merged**